Objective-C/Tests/ReplicatorTest+CustomConflict.m

```diff
@@ -734,6 +735,64 @@ - (void) testNonBlockingConflictResolver {
     AssertEqualObjects(order[1], order[2]);
 }
 
+- (void) testDoubleConflictResolutionOnSameConflicts {
+    NSString* docID = @"doc1";
+    CustomLogger* custom = [[CustomLogger alloc] init];
+    custom.level = kCBLLogLevelWarning;
+    CBLDatabase.log.custom = custom;
+    XCTestExpectation* expCCR = [self expectationWithDescription:@"wait for conflict resolver"];
+    XCTestExpectation* expSTOP = [self expectationWithDescription:@"wait for replicator to stop"];
+    NSDictionary* localData = @{@"key1": @"value1"};
+    NSDictionary* remoteData = @{@"key2": @"value2"};
+    [self makeConflictFor: docID withLocal: localData withRemote: remoteData];
+    TestConflictResolver* resolver;
+    CBLReplicatorConfiguration* pullConfig = [self config: kCBLReplicatorTypePull];
+    
+    __block int ccrCount = 0;
+    resolver = [[TestConflictResolver alloc] initWithResolver: ^CBLDocument* (CBLConflict* con) {
+        int c = ccrCount;
+        if (ccrCount++ == 0) {
+            [expCCR fulfill];
+            [NSThread sleepForTimeInterval: 2.0];
```

Comment:
I think instead of sleep, you can have another expection and wait for the expectation to fulfill. The expectation will be fulfill when getting the first document replication notification.